server/apps/archive/archive_test.py

```diff
@@ -161,10 +161,8 @@ def test_validate_schedule(self):
         date_in_past = utcnow() + timedelta(hours=-2)
         date_in_future = utcnow() + timedelta(hours=2)
 
-        ArchiveService().validate_schedule(date_in_future)
-
-        with assert_raises(SuperdeskApiError):
-            ArchiveService().validate_schedule("2015-04-27T10:53:48+00:00")
-            ArchiveService().validate_schedule(date_without_time)
-            ArchiveService().validate_schedule(time_without_date)
-            ArchiveService().validate_schedule(date_in_past)
+        validate_schedule(date_in_future)
+        self.assertRaises(SuperdeskApiError, validate_schedule, "2015-04-27T10:53:48+00:00")
+        self.assertRaises(SuperdeskApiError, validate_schedule, date_without_time)
+        self.assertRaises(SuperdeskApiError, validate_schedule, time_without_date)
+        self.assertRaises(SuperdeskApiError, validate_schedule, date_in_past)
```

Comment:
It seems that this method tries to test to cover to many test cases. Since being a unit test, I suggest splitting it into multiple test cases with each of them testing only a single edge case.